src/placeos-driver.cr

```diff
@@ -359,19 +359,12 @@ abstract class PlaceOS::Driver
 
             {{method.name.stringify}} => {
               {% for arg in args %}
-                {{arg.name.stringify}} => [
-                  {% if !arg.restriction.is_a?(Union) && arg.restriction.resolve < ::Enum %}
-                    "String",
-                    {% if !arg.default_value.is_a?(Nop) %}
-                      {{arg.default_value}}.to_s
-                    {% end %}
-                  {% else %}
-                    {{arg.restriction.stringify}},
-                    {% if !arg.default_value.is_a?(Nop) %}
-                      {{arg.default_value}}
-                    {% end %}
+                {{arg.name.stringify}} => {
+                  PlaceOS::Driver::Settings.introspect({{arg.restriction.resolve}}).to_json,
```

Comment:
What happens when `arg.restriction` is a `Nop`?

Reply:
it fails as all public methods have to specify a type, which is checked before it reaches here on line 285:
`{% raise "#{@type}##{method.name} argument `#{arg.name}` is missing a type" if arg.restriction.is_a?(Nop) %}`